index.bs

```diff
@@ -1370,77 +896,768 @@ What configuration is the device in after it resets?
+            with a {{NetworkError}} and abort these steps.
+    1. <a>Resolve</a> |promise| with |result|.
+</div>
+
+<div algorithm="USBDevice.isochronousTransferOut(endpointNumber, data, packetLengths)">
+    The <dfn for=USBDevice method>isochronousTransferOut(|endpointNumber|, |data|, |packetLengths|)</dfn> method,
+    when invoked, MUST return a new {{Promise}} |promise| and run the
+    following steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
+    1. Let |endpointAddress| be |endpointNumber| (i.e {{USBDirection/"out"}} direction).
+    1. Let |endpoint| be the result of <a>finding the endpoint</a> with
+        |endpointAddress| and <a>this</a>.
+    1. If |endpoint| is <code>null</code>, <a>reject</a> |promise| with a
+        {{NotFoundError}} and abort these steps.
+    1. If |endpoint|.{{USBEndpoint/type}} is not equal to {{USBEndpointType/"isochronous"}},
+        <a>reject</a> |promise| with an {{InvalidAccessError}} and abort these steps.
+    1. Let |result| be a new {{USBIsochronousOutTransferResult}}.
+    1. Enqueue an <a lt="isochronous transfers">isochronous OUT transfer</a> on
+        |endpoint| that will write |data| to the device, divided into
+        <code>|packetLength|.length</code> packets of
+        <code>|packetLength|[i]</code> bytes (for packets |i| from <code>0</code>
+        to <code>|packetLengths|.length - 1</code>).
+    1. For each packet |i| from <code>0</code> to <code>|packetLengths|.length -
+        1</code> the host attempts to send to the device:
+            1. Let |packet| be a new {{USBIsochronousOutTransferPacket}} and set
+                <code>|result|.{{USBIsochronousOutTransferResult/packets}}[i]</code>
+                to |packet|.
+            1. Let
+                <code>|packet|.{{USBIsochronousOutTransferPacket/bytesWritten}}</code>
+                be the amount of data successfully sent to the device as part of this
+                packet.
+            1. If the transfer ends because |endpoint| is stalled set
+                <code>|packet|.{{USBIsochronousOutTransferPacket/status}}</code> to
+                {{"stall"}}.
+            1. If the device acknowledges the complete transfer set
+                <code>|packet|.{{USBIsochronousOutTransferPacket/status}}</code> to
+                {{"ok"}}.
+            1. If the transfer fails for any other reason <a>reject</a> |promise|
+                with a {{NetworkError}} and abort these steps.
+    1. <a>Resolve</a> |promise| with |result|.
+</div>
+
+<div algorithm="USBDevice.reset()">
+    The <dfn for=USBDevice method>reset()</dfn> method,
+    when invoked, MUST return a new {{Promise}}
+    |promise| and run the following steps <a>in parallel</a>:
+    1. <a>Check if the device is configured</a> with <a>this</a> and |promise|
+        and abort these steps if promise is rejected.
```

Comment:
```suggestion
        and abort these steps if |promise| is rejected.
```

Reply:
Updated in https://github.com/WICG/webusb/pull/212/commits/5aaf696938ed1dc073f8cc8da8c809e9e5019f41.